vmware-tanzu/nsx-operator
# update NetworkConfiguration CR

# Create VPC with new VPC API model

1. Create a Project `wenqi-test` in NSX UI



2. Create a namespace

```
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    nsx.vmware.com/vpc_network_config: wenqi-1
  name: wenqi-1
```

3. Create a VPC with a new VPC CRD yaml

```
apiVersion: nsx.vmware.com/v1alpha1
kind: VPCNetworkConfiguration
metadata:
  name: wenqi-1
spec:
  privateIPs: [10.30.10.0/24]
  defaultSubnetSize: 24
  vpcServiceProfile: /orgs/default/projects/wenqi-test/vpc-service-profiles/default
  nsxProject: /orgs/default/projects/wenqi-test
  vpcConnectivityProfile: /orgs/default/projects/wenqi-test/vpc-connectivity-profiles/default
```

4. Check VPC resources in the NSX:

```
{
    "results": [
        {
            "vpc_connectivity_profile": "/orgs/default/projects/wenqi-test/vpc-connectivity-profiles/default",
            "vpc_service_profile": "/orgs/default/projects/wenqi-test/vpc-service-profiles/default",
            "load_balancer_vpc_endpoint": {
                "enabled": true
            },
            "private_ipv4_blocks": [
                "/orgs/default/projects/wenqi-test/infra/ip-blocks/e13a3582-bf1f-42ec-bc2a-f36f7e26a353_10.30.10.0"
            ],
            "ip_address_type": "IPV4",
            "short_id": "F3na1AVk",
            "resource_type": "Vpc",
            "id": "e13a3582-bf1f-42ec-bc2a-f36f7e26a353",
            "display_name": "vpc-b7b472a5-677d-4515-990f-c4bf9c1a0aa2--wenqi-1",
            "tags": [
                {
                    "scope": "nsx-op/cluster",
                    "tag": "b7b472a5-677d-4515-990f-c4bf9c1a0aa2"
                },
                {
                    "scope": "nsx-op/version",
                    "tag": "1.0.0"
                },
                {
                    "scope": "nsx-op/namespace",
                    "tag": "wenqi-1"
                },
                {
                    "scope": "nsx-op/namespace_uid",
                    "tag": "e13a3582-bf1f-42ec-bc2a-f36f7e26a353"
                }
            ],
            "path": "/orgs/default/projects/wenqi-test/vpcs/e13a3582-bf1f-42ec-bc2a-f36f7e26a353",
            "relative_path": "e13a3582-bf1f-42ec-bc2a-f36f7e26a353",
            "parent_path": "/orgs/default/projects/wenqi-test",
            "unique_id": "a0e79d9e-cbf0-4ff7-ace2-f7a813543901",
            "realization_id": "a0e79d9e-cbf0-4ff7-ace2-f7a813543901",
            "owner_id": "4447cce7-65b6-4fa0-b42c-38e9846747ac",
            "marked_for_delete": false,
            "overridden": false,
            "_system_owned": false,
            "_protection": "REQUIRE_OVERRIDE",
            "_create_time": 1721094651610,
            "_create_user": "wcp-cluster-user-b7b472a5-677d-4515-990f-c4bf9c1a0aa2-eb929be3-7d77-4b34-bf75-586933753a75",
            "_last_modified_time": 1721094652959,
            "_last_modified_user": "wcp-cluster-user-b7b472a5-677d-4515-990f-c4bf9c1a0aa2-eb929be3-7d77-4b34-bf75-586933753a75",
            "_revision": 2
        }
    ],
    "result_count": 1,
    "sort_by": "display_name",
    "sort_ascending": true
}
```

![image](https://github.com/user-attachments/assets/37104f26-065a-4bea-a9b1-f9497be4e629)
